runtime/store.go

```diff
@@ -90,3 +93,72 @@ func (store coreKVStore) Iterator(start, end []byte) (store.Iterator, error) {
 func (store coreKVStore) ReverseIterator(start, end []byte) (store.Iterator, error) {
 	return store.kvStore.ReverseIterator(start, end), nil
 }
+
+// Adapter
+var _ storetypes.KVStore = kvStoreAdapter{}
+
+type kvStoreAdapter struct {
+	store store.KVStore
+}
+
+func (kvStoreAdapter) CacheWrap() storetypes.CacheWrap {
+	panic("unimplemented")
+}
+
+func (kvStoreAdapter) CacheWrapWithTrace(w io.Writer, tc storetypes.TraceContext) storetypes.CacheWrap {
+	panic("unimplemented")
+}
+
+func (kvStoreAdapter) GetStoreType() storetypes.StoreType {
+	panic("unimplemented")
+}
+
+func (s kvStoreAdapter) Delete(key []byte) {
+	err := s.store.Delete(key)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func (s kvStoreAdapter) Get(key []byte) []byte {
+	bz, err := s.store.Get(key)
+	if err != nil {
+		panic(err)
+	}
+	return bz
+}
+
+func (s kvStoreAdapter) Has(key []byte) bool {
+	has, err := s.store.Has(key)
+	if err != nil {
+		panic(err)
+	}
+	return has
+}
+
+func (s kvStoreAdapter) Set(key []byte, value []byte) {
+	err := s.store.Set(key, value)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func (s kvStoreAdapter) Iterator(start []byte, end []byte) dbm.Iterator {
+	it, err := s.store.Iterator(start, end)
+	if err != nil {
+		panic(err)
+	}
+	return it
+}
+
+func (s kvStoreAdapter) ReverseIterator(start []byte, end []byte) dbm.Iterator {
+	it, err := s.store.ReverseIterator(start, end)
+	if err != nil {
+		panic(err)
+	}
+	return it
+}
+
+func KVStoreAdapter(store store.KVStore) storetypes.KVStore {
+	return &kvStoreAdapter{store}
+}
```

Comment:
Not sure where to put this adapter, accepting suggestions :D

Reply:
why is this needed? the new interface returns errors can we use that or?

Reply:
Because there's the prefix store and the query.Paginate that take in the KVStore from /store/types instead of core. Once we migrate all modules we could remove this adapter and modify the other methods to take in the core KVStore.

Reply:
So yes, because the new interface returns errors so it's not compatible

Reply:
makes sense thank you